xbmc/guilib/Texture.cpp

```diff
@@ -111,32 +111,24 @@ void CBaseTexture::Update(unsigned int width, unsigned int height, unsigned int
   if (pixels == NULL)
     return;
 
-  if (format & XB_FMT_DXT_MASK && !g_Windowing.SupportsDXT())
-  { // compressed format that we don't support
-    Allocate(width, height, XB_FMT_A8R8G8B8);
-    CDDSImage::Decompress(m_pixels, std::min(width, m_textureWidth), std::min(height, m_textureHeight), GetPitch(m_textureWidth), pixels, format);
-  }
-  else
-  {
-    Allocate(width, height, format);
+  Allocate(width, height, format);
```

Comment:
In my view, some error handling here might be useful, as users will almost certainly attempt to load old .dds files or skins through hackery. i.e. check for the flag, and return early is probably enough.